lib/ansible/plugins/doc_fragments/shell_common.py

```diff
@@ -56,7 +56,7 @@ class ModuleDocFragment(object):
     vars:
       - name: ansible_async_dir
   environment:
-    type: dict
+    type: dict or list
```

Comment:
these are for modules, handled by argspec, this change should not be part of this PR

Reply:
They are (also) used by the shell plugins (https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/shell/sh.py#L15, https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/shell/powershell.py#L16, https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/shell/cmd.py#L14), and these will stop working without this change. Without this change (and the one in `shell_windows.py`) CI looks like this: https://app.shippable.com/github/ansible/ansible/runs/174226/summary/console

Reply:
(Also I'm not aware of any module using these docs fragments.)

Reply:
'type: dict or list' skips validation because that isn't a known type. The convention we've used for that in the past is 'type: raw'. But environment should just be a list, since it's coming from the field attribute anyway.

Reply:
I changed it to `type: list` as discussed in the meeting.